### DISCUSSION TITLE
Problem with connection?

### DISCUSSION THREAD

**Discussion by jfornander** [echavet/MitsubishiCN105ESPHome#158] [Q&A] [answered]
I have a Mitsubishi Electric LN35 HERO and I have connected an ESP8266 to the pump. Used the "full configuration" and the ESP is connected but not without problem. I can turn on the pump and choose different modes but the current temperature is N/A (-1 in HA). Changes like temperature/fan speed seems to work (pump is changing) but all of a sudden the ESP displays "OFF" and the pump goes back to some "start state". Log is displaying:

21:08:16 | [W] | [CN105:092] | could not write as asked, because UART is not connected
21:08:16 | [I] | [CN105:008] | setupUART() with baudrate  2400
21:08:16 | [W] | [CN105:094] | delaying packet writing because we need to reconnect first...
21:08:16 | [I] | [CN105:386] | --> Heatpump did reply: connection success! <--
21:08:25 | [W] | [CN105:009] | Cycle timeout, reseting cycle...
21:08:25 | [I] | [CYCLE:052] | 6: Cycle ended in 9.0 seconds (with timeout?: YES)
21:08:29 | [I] | [CYCLE:039] | 1: Cycle start

![image](https://github.com/user-attachments/assets/3b116377-68f9-4788-a258-0e9570ff6275)

IIt seems like the pump/ESP is disconnecting frequently? Don’t know where to start, anybody?

**Comment by echavet** [comment]
Could you look at your device diagnostic sensors in the esphome integration page and make a screen shot ?

![image](https://github.com/user-attachments/assets/54c9a72f-fbbe-4e07-8d5c-183a2d521227)

**Reply by echavet** [reply]
You can also screen the uptime connection sensor graph (going to historical view)
This graph should be linear and continuous.

Mine is continuous which means no connection loss.
![image](https://github.com/user-attachments/assets/46ff85c2-817e-43fd-8771-b6bceba305d6)

**Reply by jfornander** [reply]
This is what my diagnostic looks like last 3h. Started the pump via the esp 3h ago and it crashed that's why all the numbers got reset. 
![Screenshot_2024-09-18-09-06-47-16_c3a231c25ed346e59462e84656a70e50](https://github.com/user-attachments/assets/cfb9da80-6950-4564-bfb9-3e34056260e1)

My uptime looks like that, it crashes when I try to send commands 
![Screenshot_2024-09-18-09-11-36-40_c3a231c25ed346e59462e84656a70e50](https://github.com/user-attachments/assets/487bc44e-3050-49e4-8391-839a5af69251)

**Comment by echavet** [comment, marked as answer]
It looks like your unit does not reply to the esp requests.
Do you have any state information on the home assistant ui component ? Like temperature, mode...
Because you have 0% of complete cycle. 
I'm surprised you esp manage to get any connection response.
So either it is a bug, or your uart connection is unstable.
You should double check the wires or test with a power regulator. There are threads on this project that deal with such regulators.
You could also test with an esp32 which uart is far more reliable

**Reply by jfornander** [reply]
I switched to an ESP32 ad now it is working, thanks
